### PR TITLE
Allow custom AuthenticationProvider and AuthenticationEventPublisher when using @EnableAuthorizationServer

### PR DESCRIPTION
**Pull request by jebeaudet**
This proposes a fix for issue #620.

I've signed the contributor's agreement.

Thanks

### CODE REVIEW

**Comment by jebeaudet** [comment]
The Travis CI build failed on `ResourceServerConfigurationTests` which is irrelevant to my changed, most likely a false positive. The previous opened PR has the same failure also.

**Comment by jebeaudet** [comment]
As I said in #620, I've added another customization for a custom `AuthenticationEventPublisher` in this same PR.

**Comment by Gabology** [comment]
Any updates on this PR?

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/annotation/web/configurers/AuthorizationServerSecurityConfigurer.java [hunk 1]
```diff
@@ -128,17 +159,24 @@ public String getCheckTokenAccess() {
...
+				ClientDetailsUserDetailsService clientDetailsUserDetailsService = new ClientDetailsUserDetailsService(clientDetailsService());
+				clientDetailsUserDetailsService.setPasswordEncoder(passwordEncoder());
+				builder.userDetailsService(clientDetailsUserDetailsService)
+					.passwordEncoder(passwordEncoder());
+			} else {
+				builder.userDetailsService(new ClientDetailsUserDetailsService(clientDetailsService()));
+			}
+		} else { 
+			for(AuthenticationProvider provider: authenticationProviders) {
+				builder.authenticationProvider(provider);
```
Comment (jgrandja) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/annotation/web/configurers/AuthorizationServerSecurityConfigurer.java:178]: This additional configuration is not needed as you can register one or more `AuthenticationProvider`(s) via `HttpSecurity.authenticationProvider()`.

**Pull request closed without merging**